src/Bedrock.Framework.Experimental/Protocols/Tls/BulkCiphers/BulkCipher.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Tls.BulkCiphers
+{
+    internal abstract class BulkCipher : IDisposable
+    {
+        protected const int AdditionalInfoHeaderSize = 13;
+        protected ulong _sequenceNumber;
+        protected BulkCipherKey _key;
+        protected int _blockShifter;
+
+        public int Overhead => _key.TagSize + sizeof(ulong);
+
+        public abstract void Decrypt(PipeWriter pipeWriter, ReadOnlySequence<byte> cipherText, TlsFrameType recordType, TlsProtocolVersion tlsVersion);
```

Comment:
Why `PipeWriter` an not `IBufferWriter<byte>`?

Reply:
Changed to IBufferWriter